### PR TITLE
Removed FPS cap of 60 by disabling vsync

### PR DESCRIPTION
**Pull request by mxple**

### CODE REVIEW

File: src/engine/core/Application.cpp [hunk 1]
```diff
@@ -85,6 +85,11 @@ namespace lei3d
 		}
 		glfwMakeContextCurrent(m_Window);
 
+		// Removes 60 fps cap set by GLFW without having to use a single buffer
+		// Apparently vendor/GPU specific if it works or not
+		// https://stackoverflow.com/questions/50412575/is-there-a-way-to-remove-60-fps-cap-in-glfw 
+		glfwSwapInterval(0);
```
Comment (elsien-dev) [inline src/engine/core/Application.cpp:91]: I'm not sure if we want to disable vsync. The fps cap problem was related to the implementation of FrameTick. If we set m_DesiredFPS to 60 it actually ran at 30 fps, so we had to hack it by setting it to 120 to get 60 fps.

We could add an option to enable/disable vsync though.

Comment (mxple) [reply src/engine/core/Application.cpp:91]: ah ok, i thought the desired fps was 120.

**Comment by kevinsadi** [comment]
@Mxple If you also wanted to investigate the framerate bug && revert the vsync change, we could work towards merging this branch in. We could also close this PR and create a new one.

**Comment by mxple** [comment]
@kevinsadi What's the framerate bug? We can close this PR and open a new one

**Comment by kevinsadi** [comment]
@Mxple so currently the fps is capped to half of what we actually specify in the Application.hpp file, we're not sure why, because our sleep duration seems to be using the right equation. This is the relevant code in Application.cpp:


```
		// Sync FPS to desired value
		const float currentDrawTime = (float)glfwGetTime() - currentTime;
		const float sleepTime = 1.0f / m_DesiredFPS - currentDrawTime;

		typedef std::chrono::duration<int, std::milli> ms;

		const ms sleepDuration = ms((int)(sleepTime * 1000));
		if (sleepTime > 0.0f)
		{
			// Note: Need to fix, currently doesn't cap at the right fps.
			std::this_thread::sleep_for(sleepDuration);
			//_sleep(sleepTime);    //Platform Dependent
		}
```

**Comment by kevinsadi** [comment]
Bug will be investigated in a new PR

**Pull request closed without merging**